src/lib/aws/construct/static-asset-deployment/main.ts

```diff
@@ -14,49 +16,92 @@ import { StaticAssetDeploymentProps } from './types'
  * class CustomConstruct extends StaticAssetDeployment {
  *   constructor(parent: Construct, id: string, props: StaticAssetDeploymentProps) {
  *     super(parent, id, props)
- *     this.props = props
- *     this.id = id
  *     this.initResources()
  *   }
  * }
  */
-export class StaticAssetDeployment extends CommonConstruct {
-  /* construct properties */
-  props: StaticAssetDeploymentProps
-  id: string
-
+export class StaticAssetDeployment extends CommonConstruct<StaticAssetDeploymentProps> {
   /* construct resources */
   staticAssetBucket: IBucket
-
-  constructor(parent: Construct, id: string, props: StaticAssetDeploymentProps) {
-    super(parent, id, props)
-    this.props = props
-    this.id = id
-  }
+  cloudfrontDistribution?: IDistribution
 
   /**
    * @summary Initialise and provision resources
    */
   public initResources() {
     this.createAssetBucket()
+    this.resolveDistribution()
     this.deployStaticAssets()
   }
 
   /**
    * @summary Create the static asset bucket
    */
   protected createAssetBucket() {
-    this.staticAssetBucket = this.s3Manager.createS3Bucket(`${this.id}-sa-bucket`, this, this.props.staticAssetBucket)
+    this.props.staticAssetBucket.bucketName = this.resolveRef(this.props.staticAssetBucket.bucketName)
```

Comment:
I'm not clear why we need to resolveRef to get the bucketName. If the bucket props uses `existingBucket: true` that should just resolve the bucket instead of create a bucket. Code is here https://github.com/gradientedge/cdk-utils/blob/main/src/lib/aws/services/simple-storage-service/main.ts#L93:L94